lgc/patch/FragColorExport.cpp

```diff
@@ -99,224 +99,170 @@ namespace lgc {
 FragColorExport::FragColorExport(LLVMContext *context) : m_context(context) {
 }
 
+// =====================================================================================================================
+// Split a possible vector value, if there are less than 4 elements, append with null value.
```

Comment:
Consider making it invalid to pass a non-vector value and assert for that. Also check that the number of vector elements is as expected (`4` or fewer?).

Reply:
I know handling only vector type makes the function more clean. but then we have to handle scalar type separately. especially padding the null values needs a loop, which I don't want to repeat that piece of code.

Reply:
Ack. I think this makes sense in this case.
But should we assert on the vector length?

Reply:
yes, I think we need to assert the vector length. fixed locally.